src/server.coffee

```diff
@@ -591,3 +597,25 @@ else
           type: 'restart-all'
       , 2000
     done()
+
+  # function to return process uptimes
+  exports.getUptime = (callback) ->
+
+    if cluster.isMaster
+      # send reponse back to API request
+      uptime =
+        master: process.uptime()
+      callback null, uptime
+    else
+      # send request to master
+      process.send
+        type: 'get-uptime'
+
+      # listen for response from master
+      process.once 'message', (uptime) ->
+        if uptime.type is 'get-uptime'
```

Comment:
The main issue I'm thinking of is that the `.once` off message listener will be cancelled if it gets a message that it isn't expecting (ie. a restart message, or something else in the future). So, what about we do something like this:

```
# listen for response from master
setupListener = ->
  process.once 'message', (uptime) ->
    if uptime.type is 'get-uptime'
      uptime =
        master: uptime.masterUptime

      # send reponse back to API request
      callback null, uptime
    else
      setupListener()

setupListener()
```

This will resetup the listener if we don't get the message we were expecting.